src/macro.ts

```diff
@@ -0,0 +1,132 @@
+// const printAST = require('ast-pretty-print')
+import type babelCore from '@babel/core'
+import {createMacro} from 'babel-plugin-macros'
+import type {MacroHandler} from 'babel-plugin-macros'
+import {getReplacement} from './helpers'
+
+const prevalMacros: MacroHandler = function prevalMacros({
+  references,
+  state,
+  babel,
+}) {
+  const fileOpts = state.file.opts
+  references.default.forEach(referencePath => {
+    const path = referencePath as babelCore.NodePath<babelCore.types.Identifier>
+    const targetPath = path.parentPath
+    if (targetPath.type === 'TaggedTemplateExpression') {
+      asTag(
+        targetPath as babelCore.NodePath<babelCore.types.TaggedTemplateExpression>,
```

Comment:
It was highlighted in our PR that what is being passed here is different to the original version, however we think it is functionally the same but don't know enough about writing Babel plugins to be sure. Our discussion [starts here](https://github.com/OpenSourceRaidGuild/babel-plugin-preval/pull/16#issuecomment-774316499) if you want more context.

We're happy to take comments and suggestions on this.

Reply:
I think we can rely on the tests for this one. Thanks!